code/modules/mob/living/carbon/xenomorph/castes/warrior/abilities_warrior.dm

```diff
@@ -262,22 +376,60 @@
 	var/datum/limb/L = get_limb(target_zone)
 
 	if (!L || (L.limb_status & LIMB_DESTROYED))
-		return
+		to_chat(X, "<span class='xenodanger'>We can't punch that which doesn't exist!</span>")
+		return FALSE
+
+	var/stagger_stacks = 3
+	var/slowdown_stacks = 3
+	var/punch_description = "powerful"
+	var/S = pick('sound/weapons/punch1.ogg','sound/weapons/punch2.ogg','sound/weapons/punch3.ogg','sound/weapons/punch4.ogg')
```

Comment:
Unneeded var just use it direct

Reply:
This was old code; I just moved it, but sure.